src/tools/openpgp-tool.c

```diff
@@ -152,37 +193,173 @@ static void show_version(void)
 	fprintf(stderr,
 		"openpgp-tool - OpenPGP card utility version " PACKAGE_VERSION "\n"
 		"\n"
-		"Copyright (c) 2012 Peter Marschall <peter@adpm.de>\n"
+		"Copyright (c) 2012-18 Peter Marschall <peter@adpm.de>\n"
 		"Licensed under LGPL v2\n");
 }
 
 
+/* prettify hex */
+static char *prettify_hex(u8 *data, size_t length, char *buffer, size_t buflen)
+{
+	if (data != NULL) {
+		int r = sc_bin_to_hex(data, length, buffer, buflen, ':');
+
+		if (r == SC_SUCCESS) {
+			char *ptr;
+
+			/* upper-case the hex-ified string */
+			for (ptr = buffer; *ptr != '\0'; ptr++)
+				*ptr = toupper(*ptr);
+
+			return buffer;
+		}
+	}
+	return NULL;
+}
+
+
+/* prettify algorithm parameters */
+static char *prettify_algorithm(u8 *data, size_t length)
+{
+	if (data != NULL && length >= 1) {
+		static char result[64];	/* large enough */
+
+		if (data[0] == 0x01 && length >= 5) {		/* RSA */
+			unsigned short modulus = (data[1] << 8) + data[2];
+			snprintf(result, sizeof(result), "RSA%u", modulus);
+			return result;
+		}
+		else if (data[0] == 0x12) {			/* ECDH */
+			strcpy(result, "ECDH");
+			return result;
+		}
+		else if (data[0] == 0x13) {			/* ECDSA */
+			strcpy(result, "ECDSA");
+			return result;
+		}
+	}
+	return NULL;
+}
+
+
+/* prettify date/time */
+static char *prettify_date(u8 *data, size_t length)
+{
+	if (data != NULL && length == 4) {
```

Comment:
I cannot use `sc_asn1_print_utctime()` here, because the `prettify_...()` functions prepare a string - either for printing or for setting environment variables - while `sc_asn1_print_utctime()` wants to print directly.